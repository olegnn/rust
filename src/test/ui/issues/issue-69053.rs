struct r#struct<r#fn>;
//~^ WARNING should have an upper camel case name
//~^^ WARNING should have an upper camel case name
//~^^^ ERROR parameter `fn` is never used

fn main() {}
