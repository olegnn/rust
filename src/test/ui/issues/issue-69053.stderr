warning: type `struct` should have an upper camel case name
  --> $DIR/issue-69053.rs:1:8
   |
LL | struct r#struct<r#fn>;
   |        ^^^^^^^^ help: convert the identifier to upper camel case: `Struct`
   |
   = note: `#[warn(non_camel_case_types)]` on by default

warning: type parameter `fn` should have an upper camel case name
  --> $DIR/issue-69053.rs:1:17
   |
LL | struct r#struct<r#fn>;
   |                 ^^^^ help: convert the identifier to upper camel case: `Fn`

error[E0392]: parameter `fn` is never used
  --> $DIR/issue-69053.rs:1:17
   |
LL | struct r#struct<r#fn>;
   |                 ^^^^ unused parameter
   |
   = help: consider removing `r#fn`, referring to it in a field, or using a marker such as `std::marker::PhantomData`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0392`.
